packages/table/CHANGELOG.md

```diff
@@ -1,5 +1,347 @@
+                          );
+                        })}
+                      </Row>
+                    ))}
+                </Row>
+              ))}
+          </Row>
+        );
+      })}
+    ```
+
+    **After**:
+
+    This will render both rows and subrows. No extra checks are needed.
+
+    ```
+    {rows.map((row: LeafyGreenTableRow<Person>) => {
+      return (
+        <Row row={row} key={row.id}>
+          {row
+            .getVisibleCells()
+            .map((cell: LeafyGreenTableCell<Person>) => {
+              return (
+                <Cell key={cell.id} cell={cell}>
+                  {flexRender(
+                    cell.column.columnDef.cell,
+                    cell.getContext(),
+                  )}
+                </Cell>
+              );
+            })}
+        </Row>
+      );
+    })}
+    ```
+
+  ### `ExpandedContent` component
+
+  - `row` is a required prop if using `useLeafyGreenTable` or `useLeafyGreenVirtualTable`.
+  - `virtualRow` is a required prop if using `useLeafyGreenVirtualTable`.
+  - Expanded content is included in the row object as a sibling of its parent row. You will need to check if a row is expanded content using `row.isExpandedContent`.
+
+    **Before**:
+
+    Expanded content is rendered as a child of `Row` and we check for `row.original.renderExpandedContent`
+
+    ```
+    {rows.map((row: LeafyGreenTableRow<Person>) => {
+      return (
+        <Row key={row.id} row={row}>
+          {row
+            .getVisibleCells()
+            .map((cell: LeafyGreenTableCell<Person>) => {
+              return (
+                <Cell key={cell.id}>
+                  {flexRender(
+                    cell.column.columnDef.cell,
+                    cell.getContext(),
+                  )}
+                </Cell>
+              );
+            })}
+          // renders expanded content
+          {row.original.renderExpandedContent && (
+            <ExpandedContent row={row} />
+          )}
+        </Row>
+      );
+    })}
+    ```
+
+    **After**:
+
+    Expanded content is rendered as a sibling of `Row`. Instead of checking for `row.original.renderExpandedContent`, we check for `row.isExpandedContent`.
+
+    ```
+    {rows.map((row: LeafyGreenTableRow<Person>) => {
+      const isExpandedContent = row.isExpandedContent ?? false;
+
+      return (
+        <Fragment key={row.id}>
+          {!isExpandedContent && (
+            <Row row={row}>
+              {row.getVisibleCells().map(cell => {
+                return (
+                  <Cell key={cell.id} id={cell.id} cell={cell}>
+                    {flexRender(
+                      cell.column.columnDef.cell,
+                      cell.getContext(),
+                    )}
+                  </Cell>
+                );
+              })}
+            </Row>
+          )}
+          // renders expanded content
+          {isExpandedContent && <ExpandedContent row={row} />}
+        </Fragment>
+      );
+    })}
```

Comment:
I wonder if there's any particular reason for wrapping these in a `Fragment`, alternatively - this seem simpler to me:

```suggestion
    {rows.map((row: LeafyGreenTableRow<Person>) => 
      row.isExpandedContent ? (
        <ExpandedContent row={row} key={row.id} />
      ) : (
        <Row row={row} key={row.id}>
          {row.getVisibleCells().map(cell => {
            return (
              <Cell key={cell.id} id={cell.id} cell={cell}>
                {flexRender(
                  cell.column.columnDef.cell,
                  cell.getContext(),
                )}
              </Cell>
            );
          })}
        </Row>
      )
    )}
```

Reply:
If I remember correctly, without the fragment, virtualized tables with opened expanded rows would start to get jumpy on scroll.